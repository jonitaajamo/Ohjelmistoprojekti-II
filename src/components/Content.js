import React, { Component } from "react";
import Map from "./content/Map";
import Currency from "./content/Currency";
import StockMarket from "./content/StockMarket";
import Portfolio from "./content/Portfolio";
import News from "./content/News";

class Content extends Component {
  render() {

    let contentStyle = {
      width: "99%",
      margin: "auto"
    }

    return (
      <div style={contentStyle} className="tile is-ancestor">
        <div className="tile is-vertical is-8">
          <div className="tile is-parent">
            <Map />
          </div>
          <div className="tile is-parent">
            <article className="tile is-child notification is-danger">
              <p className="subtitle">Timeline placeholder</p>
            </article>
          </div>
          <div className="tile is-parent">
              <Portfolio />
            </div>
          <div className="tile">
            <div className="tile is-parent is-vertical">
              <Currency />
              <StockMarket />
            </div>
            
          </div>
        </div>
        <div className="tile is-parent">
          <News />
        </div>
      </div>
    );
  }
}

export default Content;
