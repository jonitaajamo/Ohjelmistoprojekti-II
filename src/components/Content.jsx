import React, { Component } from "react";
import { feature } from "topojson-client";
import Map from "./content/Map";
import Assets from "./content/Assets";
import Portfolio from "./content/Portfolio";

export default class Content extends Component {
  constructor(props) {
    super(props);
    this.state = {
      geographicalWeightData: [],
      geographyBorders: [],
      geographyNames: [],
      selectedMonth: 0,
      selectedAsset: 0,
      disableOptimization: false
    };
  }

  fetchDataForMap() {
    fetch(
      "https://raw.githubusercontent.com/tarmeli/Ohjelmistoprojekti-II/master/src/data/getGeographicalWeights-mock.json"
    )
      .then(response => response.json())
      .then(weights =>
        this.setState({
          geographicalWeightData: weights.monthlyWeights
        })
      )
      .catch(err => console.error(err));

      fetch("https://unpkg.com/world-atlas@1.1.4/world/110m.json")
      .then(response => response.json())
      .then(worldData =>
        this.setState({
          geographyBorders: feature(
            worldData,
            worldData.objects.countries
          ).features.sort((a, b) => {
            return a.id - b.id;
          })
        })
      )
      .catch(err => console.error(err));

    fetch(
      "https://raw.githubusercontent.com/tarmeli/Ohjelmistoprojekti-II/master/src/data/countryNames.json"
    )
      .then(response => response.json())
      .then(names =>
        this.setState({
          geographyNames: names.countries.sort((a, b) => {
            return a.id - b.id;
          })
        })
      )
      .catch(err => console.error(err));
  }

  componentDidMount() {
    this.fetchDataForMap();
  }

  changeMonth(event) {
    this.setState(
      {
        selectedMonth: JSON.parse(event.target.value),
        disableOptimization: true
      },
      () => {
        this.setState({
          disableOptimization: false
        });
      }
    );
  }

  changeAsset(item) {
    this.setState(
      {
        selectedAsset: item,
        disableOptimization: true
      },
      () => {
        this.setState({
          disableOptimization: false
        });
      }
    );
  }

  changeMonthOnClick(event) {
    const { selectedMonth, geographicalWeightData } = this.state;
    const weightDataLength = geographicalWeightData.length - 1;
    if (event.target.classList.contains("left") && selectedMonth > 0) {
      this.setState(
        prevState => {
          return {
            selectedMonth: prevState.selectedMonth - 1,
            disableOptimization: true
          };
        },
        () => {
          this.setState({
            disableOptimization: false
          });
        }
      );
    } else if (
      event.target.classList.contains("right") &&
      selectedMonth < weightDataLength
    ) {
      this.setState(
        prevState => {
          return {
            selectedMonth: prevState.selectedMonth + 1,
            disableOptimization: true
          };
        },
        () => {
          this.setState({
            disableOptimization: false
          });
        }
      );
    }
  }

  render() {
    const contentStyle = {
      width: "99%",
      margin: "auto"
    };

    const weightDataLength = this.state.geographicalWeightData.length - 1;

    return (
      <div style={contentStyle} className="tile is-ancestor">
        <div className="tile is-vertical is-8">
          <div className="tile is-parent">
            <Map
              geographicalWeightData={this.state.geographicalWeightData}
              geographyNames={this.state.geographyNames}
              geographyBorders={this.state.geographyBorders}
              selectedMonth={this.state.selectedMonth}
              selectedAsset={this.state.selectedAsset}
              disableOptimization={this.state.disableOptimization}
              length={weightDataLength}
              onChange={this.changeMonth.bind(this)}
              onAssetChange={this.changeAsset.bind(this)}
              month={this.state.selectedMonth}
              onMonthButtonClick={this.changeMonthOnClick.bind(this)}
            />
          </div>

          <div className="tile is-parent">
            <Portfolio />
          </div>
        </div>
        <div className="tile">
          <div className="tile is-parent is-vertical">
            <Assets />
          </div>
        </div>
      </div>
    );
  }
}
