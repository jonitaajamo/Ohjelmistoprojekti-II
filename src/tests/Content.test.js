import React from "react";
import Content from "../components/Content";

describe("Content", () => {
  it("renders correctly", () => {
    const wrapper = shallow(<Content />);
    expect(wrapper).toMatchSnapshot();
  });

  it("renders correctly with children", () => {
    const wrapper = mount(<Content />);
    expect(wrapper.find("Map").length).toEqual(1);
    expect(wrapper.find("Assets").length).toEqual(1);
  });

  it("fetches data when mounted", () => {
    const spy = jest.spyOn(Content.prototype, "fetchDataForMap");
    const wrapper = shallow(<Content />);
    expect(spy).toHaveBeenCalledTimes(1);
  });
});
