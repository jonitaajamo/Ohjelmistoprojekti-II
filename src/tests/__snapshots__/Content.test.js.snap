// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Content renders correctly 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Content />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <Map
        currency=""
        disableOptimization={false}
        fetchGeographyIdFromMap={[Function]}
        geographicalWeightData={Array []}
        geographyBorders={Array []}
        geographyId=""
        geographyNames={Array []}
        isGeographyClicked={false}
        isLoaded={false}
        length={-1}
        month={0}
        onAfterChange={[Function]}
        onAssetChange={[Function]}
        onChange={[Function]}
        onMonthButtonClick={[Function]}
        selectedAsset={0}
        selectedMonth={0}
        topInstruments={Array []}
/>,
      "style": Object {
        "margin": "10px",
      },
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "currency": "",
        "disableOptimization": false,
        "fetchGeographyIdFromMap": [Function],
        "geographicalWeightData": Array [],
        "geographyBorders": Array [],
        "geographyId": "",
        "geographyNames": Array [],
        "isGeographyClicked": false,
        "isLoaded": false,
        "length": -1,
        "month": 0,
        "onAfterChange": [Function],
        "onAssetChange": [Function],
        "onChange": [Function],
        "onMonthButtonClick": [Function],
        "selectedAsset": 0,
        "selectedMonth": 0,
        "topInstruments": Array [],
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <Map
          currency=""
          disableOptimization={false}
          fetchGeographyIdFromMap={[Function]}
          geographicalWeightData={Array []}
          geographyBorders={Array []}
          geographyId=""
          geographyNames={Array []}
          isGeographyClicked={false}
          isLoaded={false}
          length={-1}
          month={0}
          onAfterChange={[Function]}
          onAssetChange={[Function]}
          onChange={[Function]}
          onMonthButtonClick={[Function]}
          selectedAsset={0}
          selectedMonth={0}
          topInstruments={Array []}
/>,
        "style": Object {
          "margin": "10px",
        },
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "class",
        "props": Object {
          "currency": "",
          "disableOptimization": false,
          "fetchGeographyIdFromMap": [Function],
          "geographicalWeightData": Array [],
          "geographyBorders": Array [],
          "geographyId": "",
          "geographyNames": Array [],
          "isGeographyClicked": false,
          "isLoaded": false,
          "length": -1,
          "month": 0,
          "onAfterChange": [Function],
          "onAssetChange": [Function],
          "onChange": [Function],
          "onMonthButtonClick": [Function],
          "selectedAsset": 0,
          "selectedMonth": 0,
          "topInstruments": Array [],
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
